documentation/puppet-api/v3/all_environment_modules.json

```diff
@@ -0,0 +1,8 @@
+{
+  "$schema":     "http://json-schema.org/draft-04/schema#",
+  "title":       "All Environment Modules",
+  "description": "An array of information about all the modules in all Puppet code environments",
+  "type":        "array",
+  "properties": ["Environment Modules"],
+  "additionalProperties": false
+}
```

Comment:
If not, then, I guess I'd either prefer to see a single JSON schema which represents both the single and array formats - reusing the single item format to avoid duplication - or two separate full docs which have the duplication.  Either way, I think it will be most helpful to people to have the schema(e) capable of validating the different resource content which can come from the endpoint.